### PR TITLE
Create CRUD functions with test cases outlined in BankAccount

### PR DESCRIPTION
**Pull request by mfdeveloper508**
Create CRUD functions with test cases outlined in BankAccount and support Micro-deposit endpoint.

Create Bank Account is not working.

### CODE REVIEW

**Review by wadearnold — APPROVED** [review]

File: pkg/bankAccount.go [hunk 1]
```diff
@@ -45,14 +52,152 @@ type ACHStatusUpdates struct {
...
+		return err
+	}
+
+	switch statusCode {
+	case http.StatusOK:
+		return nil
+	case http.StatusUnauthorized:
+		return ErrAuthCreditionalsNotSet
+	case http.StatusUnprocessableEntity:
+		log.Println("UnprocessableEntity")
```
Comment (adamdecaf) [inline pkg/bankAccount.go:200]: Is the plan to consolidate these into a helper?

File: pkg/bankAccount.go [hunk 2]
```diff
@@ -45,14 +52,152 @@ type ACHStatusUpdates struct {
...
+	}
+	return nil
+}
 
 // MicroDepositConfirm confirms a micro deposit verification for the given bank account
+func (c Client) MicroDepositConfirm(accountID string, bankAccountID string, amounts []int) error {
+	url := fmt.Sprintf("%s/%s/%s/micro-deposits", baseURL, fmt.Sprintf(endpoint, accountID), bankAccountID)
+
+	_, statusCode, err := GetHTTPResponse(c, http.MethodPut, url, map[string][]int{"amounts": amounts})
```
Comment (adamdecaf) [inline pkg/bankAccount.go:189]: Is there a struct we can use from the OpenAPI doc that has the request body?

File: pkg/utils.go [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+	PublicKey string `yaml:"public_key,omitempty"`
+	SecretKey string `yaml:"secret_key,omitempty"`
+	Domain    string `yaml:"domain,omitempty"`
+}
+
+func readConfig() (Credentials, error) {
+	var cred Credentials
+
+	// Read the config file
+	data, err := os.ReadFile("../config.yaml")
```
Comment (adamdecaf) [inline pkg/utils.go:23]: Why a yaml config with a hardcoded path?

File: pkg/utils.go [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+func GetHTTPResponse(c Client, method string, url string, data any) ([]byte, int, error) {
+	// Make an HTTP request
+	var req *http.Request
+	if data != nil {
+		payload, err := json.Marshal(data)
+		if err != nil {
+			return nil, 0, err
+		}
+
+		req, err = http.NewRequest(method, url, bytes.NewBuffer(payload))
```
Comment (adamdecaf) [inline pkg/utils.go:46]: We should Marshal to a `bytes.Buffer` and pass it into NewRequest.

**Pull request merged**